bsp4j/src/main/java/ch/epfl/scala/bsp4j/ScalaExtension.xtend

```diff
@@ -170,3 +171,36 @@ class ScalaTestSuiteSelection {
     this.tests = tests
   }
 }
+
+@JsonRpcData
+class ScalaDiagnostic {
```

Comment:
I am 💯% with you that it's _not a great_ name, **but** there's already a very strong precedence set by `buildTarget.data` ecosystem, which basically uses the same convention of stuff JSON into `data` and declaring the kind using `dataKind`, so in that senses, `data` is not a total free-for-all, but more like a dynamic switch.
In that world, the naming convention seems to be `(dataKind + "BuildTarget").capitalize`, i.e. `ScalaBuildTarget` for `scala`, `JvmBuildTarget ` for `jvm`, and `CppBuildTarget` for `cpp` etc. I wish it said `ScalaBuildTargetData`, but alas!

Coming back here, while I too do not love the name, I also think we should be consistent here and call this `ScalaDiagnostic`, and definitely not `ScalaActions` because part of the reasons we want this to be a JSON object is so we can add more fields in the future, potentially, and names here matter as it would travel to places like sbt code base where compatibility is paramount.

Reply:
Ah, good point since this could indeed contain more stuff in the future. Would `ScalaData` be consistent and still be flexible enough to allow for future things to be added?

Reply:
> Would `ScalaData` be consistent and still be flexible enough to allow for future things to be added?

Almost all structures in BSP can contain `data`, not just `BuildTarget`, but also `InitializeBuildParams`, `DependencyModule`, `CompileResult` etc, so `ScalaData` might be too vague.

Reply:
`ScalaDiagnosticData` :trollface: joking, we're going to end up going full circle here and land back at `ScalaDiagnostic`. No worries 👍🏼